packages/govern-core/contracts/GovernRegistry.sol

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier:    GPL-3.0
+ */
+
+pragma solidity 0.6.8;
+
+import "erc3k/contracts/ERC3000.sol";
+import "erc3k/contracts/ERC3000Executor.sol";
+import "erc3k/contracts/ERC3000Registry.sol";
+
+import "@aragon/govern-contract-utils/contracts/erc165/ERC165.sol";
+
+contract GovernRegistry is ERC3000Registry {
+    mapping (string => bool) public nameUsed;
+
+    function register(ERC3000Executor _dao, ERC3000 _queue, string calldata _name, bytes calldata _initialMetadata) override external
+    {
+        require(!nameUsed[_name], "registry: name used");
+        require(ERC165(address(_queue)).supportsInterface(ERC3000_INTERFACE_ID), "registry: bad interface queue");
```

Comment:
Let's remove or comment this to make the registry a bit cheaper

Reply:
I will remove it in the ``_setMetadata`` contract method as well 👍